desktop/src/renderer/src/components/transfer/UploadGrid/StatusCell.tsx

```diff
@@ -0,0 +1,60 @@
+import { Box, Stack, Tooltip, Typography } from "@mui/material";
+import { Circle as ProgressIcon } from "@mui/icons-material";
+import type {
+  GridRenderCellParams,
+  GridTreeNodeWithRender,
+} from "@mui/x-data-grid";
+import type { Row } from "./FolderUploadGrid";
+
+type Props = {
+  params: GridRenderCellParams<Row, unknown, unknown, GridTreeNodeWithRender>;
+};
+
```

Comment:
Add a method that returns the metadata based on `status`. (For the follow up comment)
```suggestion
const getStatusDetails = (
  complete: boolean,
  invalidPath: boolean,
  progress: number
) => {
  if (invalidPath) {
    return {
      tooltip: "Folder path could not be found. Use the edit button to change folder path.",
      iconColor: "var(--progress-cancelled)",
      progressText: "",
    };
  }

  if (complete) {
    return {
      tooltip: "Upload complete.",
      iconColor: "var(--progress-complete)",
      progressText: `${progress}%`,
    };
  }

  return {
    tooltip: "Upload in progress.",
    iconColor: "var(--progress-incomplete)",
    progressText: progress > 0 ? `${progress}%` : "",
  };
};

```